app/core/pokemon-entity.ts

```diff
@@ -535,6 +535,28 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
       )
     }
   }
+  
+  addPermanentStats(
+    stat: { [K in keyof IPokemon]:
+      IPokemon[K] extends number ? K : never
+      }[keyof IPokemon],
+    value: number
+  ) {
+    if(this.isGhostOpponent) return
+    this.refToBoardPokemon[stat] += value
+  }
+
+  addPermanentItem(
+    item: Item,
+    remove = false
```

Comment:
confusing method name for removing, declare a `removePermanentItem` method instead